### PR TITLE
[Backport]Added condition to resolve issue in Rhel7.6alt

### PR DESCRIPTION
**Pull request by shirishaganta**
Based on issue #1406 modified the script to run correctly
Also changed Class name according to Classes naming convention

Signed-off-by: Shirisha Ganta <shiganta@in.ibm.com>

### CODE REVIEW

**Comment by shirishaganta** [comment]
Before Patch:
JOB ID     : 15083accedf6804440053110f3c74fae5fe31937
JOB LOG    : /home/avocado-fvt-wrapper-py2/results/job-2019-11-14T11.10-15083ac/job.log
 (1/3) perf_uprobe.py:PerfUprobe.test_uprobe:  PASS (8.03 s)
 (2/3) perf_uprobe.py:PerfUprobe.test_uprobe_return:  ERROR: Command 'perf record -o /var/tmp/tmpY1XkKw -e probe_uprobe_test:doit__return -- ./uprobe_test' failed.\nstdout: ''\nstderr: "event syntax error: 'probe_uprobe_test:doit__return'\n                     \\___ unknown tracepoint\n\nError:\tFile /sys/kernel/debu... (0.84 s)
 (3/3) perf_uprobe.py:PerfUprobe.test_uprobe_variable:  PASS (1.04 s)
RESULTS    : PASS 2 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 10.46 s
JOB HTML   : /home/avocado-fvt-wrapper-py2/results/job-2019-11-14T11.10-15083ac/results.html

After Patch:
JOB ID     : 9521872a2ae64905ef5d0c110a8ee1b7afa92aa8
JOB LOG    : /home/avocado-fvt-wrapper-py2/results/job-2019-11-14T15.37-9521872/job.log
 (1/3) perf_uprobe.py:PerfUprobe.test_uprobe: PASS (2.72 s)
 (2/3) perf_uprobe.py:PerfUprobe.test_uprobe_return: PASS (1.13 s)
 (3/3) perf_uprobe.py:PerfUprobe.test_uprobe_variable: PASS (1.05 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 5.46 s
JOB HTML   : /home/avocado-fvt-wrapper-py2/results/job-2019-11-14T15.37-9521872/results.html

File: perf/perf_uprobe.py [hunk 1]
```diff
@@ -72,10 +73,10 @@ def cmd_verify(self, cmd):
 
     def test_uprobe(self):
         output = self.cmd_verify('%s /usr/bin/perf main' % self.cmdProbe)
-        if 'Added new event' not in output.stderr:
+        if 'Added new event' not in output.stderr.decode("utf-8"):
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: why decode is required  i don't think it require if test run against avcoado 69 lts

File: perf/perf_uprobe.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 from avocado.utils.software_manager import SoftwareManager
 
 
-class perfUprobe(Test):
+class PerfUprobe(Test):
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: why this change

File: perf/perf_uprobe.py [hunk 3]
```diff
@@ -84,34 +85,34 @@ def test_uprobe(self):
...
-        if 'Added new event' not in output.stderr:
+        if 'Added new event' not in output.stderr.decode("utf-8"):
             self.fail("perf: probe of uprobe return failed")
         # RHEL
-        if self.distro_name == "rhel":
+        if self.distro_name == "rhel" and self.distro_version > "7":
             output = self.cmd_verify('%s__return -- ./uprobe_test'
                                      % self.recProbe)
         else:
             output = self.cmd_verify('%s -aR ./uprobe_test' % self.recProbe)
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: just one help this will against older RHEL distro ? and can this is applicable for RHEL based distro like Fedora

File: perf/perf_uprobe.py (hunk 1 above)
Comment (shirishaganta) [reply perf/perf_uprobe.py]: Yeah no need i will do changes

File: perf/perf_uprobe.py (hunk 2 above)
Comment (shirishaganta) [reply perf/perf_uprobe.py]: That is python class convention i followed here Praveen

Comment (PraveenPenguin) [reply perf/perf_uprobe.py]: yes i know but it should be documented in commit message :)  ideal way it should go as seperate commit

Comment (shirishaganta) [reply perf/perf_uprobe.py]: Oh that i donno so i sent it in the same commit

Comment (PraveenPenguin) [reply perf/perf_uprobe.py]: should be fine but please document as part of commit message :)

Comment (shirishaganta) [reply perf/perf_uprobe.py]: sure Praveen

File: perf/perf_uprobe.py (hunk 3 above)
Comment (shirishaganta) [reply perf/perf_uprobe.py]: Yes Praveen this will run for older rhel distro as well
